scheme/sevsnp/store_handler.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2025 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package sevsnp
+
+import (
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"net/url"
+
+	"github.com/veraison/corim/comid"
+	"github.com/veraison/ratsd/tokens"
+	"github.com/veraison/services/handler"
+	"github.com/veraison/services/proto"
+)
+
+// StoreHandler implements the IStoreHandler interface handler for SEVSNP scheme
+type StoreHandler struct{}
+
+// GetName returns the name of this StoreHandler instance
+func (s StoreHandler) GetName() string {
+	return fmt.Sprintf("%s-store-handler", SchemeName)
+}
+
+// GetAttestationScheme returns the attestation scheme
+func (s StoreHandler) GetAttestationScheme() string {
+	return SchemeName
+}
+
+// GetSupportedMediaTypes returns the supported media types; no-op for SEVSNP
+func (s StoreHandler) GetSupportedMediaTypes() []string {
+	return nil
+}
+
+// getRefValKey helper to compute RefVal key from CoMID value triple
+func getRefValKey(rv comid.ValueTriple, tenantID string) (string, error) {
+	m, err := measurementByUintKey(rv, mKeyMeasurement)
+	if err != nil {
+		return "", err
+	}
+
+	if m == nil {
+		return "", fmt.Errorf("measurement not found")
+	}
+
+	d := m.Val.Digests
+
+	u := url.URL{
+		Scheme: SchemeName,
+		Host:   tenantID,
+		Path:   hex.EncodeToString((*d)[0].HashValue),
+	}
+
+	return u.String(), nil
+}
+
+// SynthKeysFromRefValue constructs SEV-SNP reference value of the form
+// "SEVSNP://<tenantID>/<measurement>". The measurement
+// is unique to an attester instance and, as such, is
+// the best candidate to use as the key.
+func (s StoreHandler) SynthKeysFromRefValue(
+	tenantID string,
+	refValue *handler.Endorsement,
+) ([]string, error) {
+	var rv comid.ValueTriple
+
+	err := json.Unmarshal(refValue.Attributes, &rv)
+	if err != nil {
+		return nil, err
+	}
+
+	refValKey, err := getRefValKey(rv, tenantID)
+	if err != nil {
+		return nil, err
+	}
+
+	return []string{refValKey}, nil
+}
+
+// SynthKeysFromTrustAnchor constructs the SEV-SNP Trust Anchor key. The
+// key format is "SEVSNP://<keyname>". For example, "SEV-SNP://ARK-Milan"
```

Comment:
```suggestion
// key format is "SEVSNP://<keyname>". For example, "SEVSNP://ARK-Milan"
```

Besides, why isn’t this using the tenant ID for namespacing?

Reply:
I'm using the tenant ID for namespacing; I'll upload that shortly